File: benchmarks/language_compare/array.c
/* BENCH_ITERATIONS=5000000 BENCH_EXPECTED=461852210 */
#include <stdint.h>
#include <stdio.h>
int main(void) {
    int64_t xs[1024];
    for (int64_t i = 0; i < 1024; i++) xs[i] = (i * 17 + 3) % 1000;
    int64_t state = 1, checksum = 0;
    for (int64_t i = 0; i < 5000000; i++) {
        state = (state * 48271) % 2147483647;
        checksum = (checksum + xs[state % 1024]) % 1000000007;
    }
    printf("%lld\n", (long long)checksum);
    return 0;
}

File: benchmarks/language_compare/array.rs
// BENCH_ITERATIONS=5000000 BENCH_EXPECTED=461852210
fn main() {
    let mut xs = [0_i64; 1024];
    for i in 0..1024 { xs[i] = ((i * 17 + 3) % 1000) as i64; }
    let mut state = 1_i64;
    let mut checksum = 0_i64;
    for _ in 0..5_000_000 {
        state = (state * 48_271) % 2_147_483_647;
        checksum = (checksum + xs[(state % 1024) as usize]) % 1_000_000_007;
    }
    println!("{checksum}");
}
